### DISCUSSION TITLE
Undefined index: a in mpdf/mpdf/src/Mpdf.php on line 13570

### DISCUSSION THREAD

**Discussion by ankita-parashar-soni** [mpdf/mpdf#1492] [Q&A]
I am using Mpdf libfary to send pdf as attachment fro Magento 2 order & invoices. I am trying to send email template as attachment with that email.

 $mpdf = new \Mpdf\Mpdf(['tempDir' => $path. '/tmp']);
$mpdf->WriteHTML($emailMessage);
$pdfHtml = $mpdf->Output('', 'S');

I'm getting the notice "Undefined index: a in mpdf/mpdf/src/Mpdf.php on line 13570"

Not sure what's triggering it.

**Comment by ssieb** [comment]
Then you need to reduce the $emailMessage content until you get a minimal testcase that can reproduce it.  If it's coming from a template, it's highly likely that there's some invalid content.

**Comment by ankita-parashar-soni** [comment]
> Then you need to reduce the $emailMessage content until you get a minimal testcase that can reproduce it. If it's coming from a template, it's highly likely that there's some invalid content.

is there any way I can validate my $emailMessage for mpf?

**Comment by ssieb** [comment]
Use an html validator.  Or save the content somewhere and remove parts of it until you have the smallest chunk that still causes that message.

**Comment by ankita-parashar-soni** [comment]
> Use an html validator. Or save the content somewhere and remove parts of it until you have the smallest chunk that still causes that message.

Tried that. I fix one error, then it throws new error. The same code is working for the same email template on production. But not on my local. Only difference is production website has magento 2.3.1 and local has Magento 2.4.2.

**Reply by Klap-in** [reply]
Please reduce it further. If you have minimal testcase(s) that can reproduce the error(s), you enable others to help you. Only pointing to place is not enough for this library with quite complicated code. So there are such examples needed to ensure fixes really solve your problem and do not trigger unwanted effects.

**Reply by ssieb** [reply]
Or try a diff between the output from each version and see what changed.
